articles/ros2_nodl.md

```diff
@@ -113,43 +118,60 @@ Another is the fact that, as soon as the node is running, RCL itself (or another
 How do upstream packages specify their interface requirements?
 Through a high-level description of all the actions, parameters, services and topics, provided or required, by each node within the package.
 
-The package interface is defined in a separate [XML][xml_wiki] file and exported from the `package.xml` using [REP 149's export mechanism][rep149_export], thereby avoiding pollution of the package manifest.
+The package interface is defined in a separate [XML][xml_wiki] file with suffix `.nodl.xml`.
+This XML file is exported to the [ament index][ament_index], either manually in the case of python projects or with a helper CMake macro.
 The interface may cover only a subset of nodes in a package, as long as the nodes that _are_ covered are done so completely.
 
-Here is an example IDL for a package containing two nodes:
+Here is an example NoDL for a package containing two nodes:
 
-{% include_relative ros2_node_idl/interface_declaration.xml %}
+{% include_relative ros2_nodl/interface_declaration.xml %}
 
-Once an IDL file is written, it is exported from the package manifest:
+Once an NoDL file is written, it is exported from either `CMakeLists.txt` or `setup.py` (more details below).
+Note that several NoDL files can be exported, allowing for writing one NoDL file per node if desired.
 
-{% include_relative ros2_node_idl/package.xml %}
+### Exporting a NoDL to the Ament Index
 
-Note that several IDL files can be exported, allowing for writing an IDL file per node.
+Per the design philosophy of the [ament index][ament_index], files will be installed in two locations.
+In the case of a package named `Foo`, the NoDL file `foo.nodl.xml` should be placed in the package share directory, `share/foo/foo.nodl.xml`.
+A corresponding marker file, `share/ament_index/nodl_desc/foo`, should be created.
+This is either empty or contains the relative path to the `foo.nodl.xml` file.
 
-### Schema for `package.xml`'s export tag
+#### CMake Macro
 
-#### `interface`
+For packages using ament_cmake, the package `ament_nodl` provides the macro `nodl_export_node_description_file` which performs the above.
```

Comment:
Replace `the above` with a link to section `Exporting a NoDL to the Ament Index`. Something along the line,
```suggestion
For packages using ament_cmake, the package `ament_nodl` provides the macro `nodl_export_node_description_file` which performs the export described in [Exporting a NoDL to the Ament Index][#exporting-a-nodl-to-the-ament-index]
```

Reply:
663747e